lib/src/test/java/tech/relaycorp/relaydroid/FirstPartyEndpointTest.kt

```diff
@@ -0,0 +1,68 @@
+package tech.relaycorp.relaydroid
+
+import com.nhaarman.mockitokotlin2.any
+import com.nhaarman.mockitokotlin2.argumentCaptor
+import com.nhaarman.mockitokotlin2.eq
+import com.nhaarman.mockitokotlin2.mock
+import com.nhaarman.mockitokotlin2.verify
+import com.nhaarman.mockitokotlin2.whenever
+import kotlinx.coroutines.test.runBlockingTest
+import org.junit.Assert.assertEquals
+import org.junit.Assert.assertNull
+import org.junit.Test
+import tech.relaycorp.relaydroid.test.FirstPartyEndpointFactory
+import tech.relaycorp.relaydroid.test.ParcelDeliveryCertificates
+import tech.relaycorp.relaynet.testing.pki.KeyPairSet
+import tech.relaycorp.relaynet.wrappers.privateAddress
+import java.security.KeyPair
+
+class FirstPartyEndpointTest {
+
+    @Test
+    fun address() {
+        val endpoint = FirstPartyEndpointFactory.build()
+        assertEquals(endpoint.keyPair.public.privateAddress, endpoint.address)
+    }
+
+    @Test
+    fun register() = runBlockingTest {
+        val gateway = mock<GatewayClientI>()
+        val storage = mock<StorageImpl>()
+        Relaynet.storage = storage
+        Relaynet.gatewayClientImpl = gateway
+
+        whenever(gateway.registerEndpoint(any())).thenReturn(Pair(
+            ParcelDeliveryCertificates.PRIVATE_ENDPOINT,
+            ParcelDeliveryCertificates.PRIVATE_GW
+        ))
+
+        val endpoint = FirstPartyEndpoint.register()
+
+        val keyPairCaptor = argumentCaptor<KeyPair>()
+        verify(gateway).registerEndpoint(keyPairCaptor.capture())
+        verify(storage).setIdentityKeyPair(eq(endpoint.address), eq(keyPairCaptor.firstValue))
+        verify(storage).setIdentityCertificate(eq(endpoint.address), eq(ParcelDeliveryCertificates.PRIVATE_ENDPOINT))
+        verify(storage).setGatewayCertificate(eq(ParcelDeliveryCertificates.PRIVATE_GW))
+    }
+
+    @Test
+    fun load() = runBlockingTest {
```

Comment:
(I forgot to hit "send" :face_with_head_bandage: )

Reply:
I tried to set it up, but couldn't make it work, it was always skipping tests. I can spend some more time on it if you want, but it will only be possible for unit tests, not instrumentation tests.

Reply:
Right. If it didn't work straightaway then it's not worth trying to get it to work.